### PR TITLE
enhance jobs api

### LINKED ISSUE #994: Switch jobs API from offset to token-based pagination
Currently overwatch uses offset based pagination to iterate over the list of jobs: [link](https://github.com/databrickslabs/overwatch/blame/736ec19767159f5204e78dc00b5c803259eae806/src/main/scala/com/databricks/labs/overwatch/api/ApiMeta.scala#L293)

Offset parameter in /jobs/list API call is going to be limited to 1000, so we need to make overwatch use page_token parameter instead: https://docs.databricks.com/api/workspace/jobs/list

### PR DESCRIPTION
**Pull request by aman-db**
closes #994

### CODE REVIEW

**Comment by sriram251-code** [comment]
please add the PR proofs

**Comment by sriram251-code** [comment]
please give access to the PR proofs

**Comment by sriram251-code** [comment]
Run using 7201 and check the job counts run using this fix and check the job counts. Make sure jobs api are paginating.

File: src/main/scala/com/databricks/labs/overwatch/api/ApiMeta.scala [hunk 1]
```diff
@@ -293,24 +293,17 @@ class JobListApi extends ApiMeta {
...
   setIsDerivePaginationLogic(true)
 
   private[overwatch] override def hasNextPage(jsonObject: JsonNode): Boolean = {
     jsonObject.get(paginationKey).asBoolean()
   }
 
   private[overwatch] override def getPaginationLogic(jsonObject: JsonNode, requestMap: Map[String, String]): Map[String, String] = {
-    val limit = Integer.parseInt(requestMap.get("limit").get)
-    var offset = Integer.parseInt(requestMap.get("offset").get)
-    val expand_tasks = requestMap.get("expand_tasks").get
```
[the remark this answers is not in the document]
Comment (aman-db) [reply src/main/scala/com/databricks/labs/overwatch/api/ApiMeta.scala:305]: @oleksiidav we are not removing expand_tasks, this is already an existing key in requestMap, we have just removed the redundant code from the function.

**Comment by aman-db** [comment]
> Run using 7201 and check the job counts run using this fix and check the job counts. Make sure jobs api are paginating.

done

**Review by sriram251-code — APPROVED** [review]

**Review by oleksiidav — APPROVED** [review]
LGTM - let's merge to make sure we make it for the connectors migration deadline in 10 days

**Comment by oleksiidav** [comment]
CC @aman-db

**Review by gueniai — APPROVED** [review]

**Pull request merged**